### PR TITLE
Update network interface

### LINKED ISSUE #18: Make network broadcast trial be able to be stopped

### LINKED ISSUE #19: Add 'check_broadcast_status(token)` for network interface
The node operator would check how well his message is broadcasted, and decide whether to turn off his node.
The return value could be
1. Number of successfully relayed nodes
2. Some kind of a 'score' that considers how well the relayed nodes will actually relay well, according to their past activities.

I think the option 1 will be sufficient for now.

### PR DESCRIPTION
**Pull request by junha1**
fixes #19 
fixes #18

### CODE REVIEW

File: network/src/lib.rs [hunk 1]
```diff
@@ -20,7 +29,11 @@ pub trait AuthorizedNetwork {
     where
         Self: Sized;
     /// Broadcasts a message to the network, after signed by the key given to this instance.
-    async fn broadcast(&self, message: &[u8]) -> Result<(), String>;
+    async fn broadcast(&self, message: &[u8]) -> Result<BroadcastToken, String>;
+    /// Stops a currently broadcasting message.
+    async fn stop_braodcast(&self, token: BroadcastToken) -> Result<(), String>;
```
Comment (lemonholic) [inline network/src/lib.rs:34]: typo

File: network/src/propagation_network/mod.rs [hunk 2]
```diff
@@ -32,9 +33,18 @@ impl AuthorizedNetwork for PropagationNetwork {
     {
         unimplemented!("not implemented");
     }
-    async fn broadcast(&self, _message: &[u8]) -> Result<(), String> {
+    async fn broadcast(&self, _message: &[u8]) -> Result<BroadcastToken, String> {
         unimplemented!("not implemented");
     }
+    async fn stop_braodcast(&self, _token: BroadcastToken) -> Result<(), String> {
```
Comment (lemonholic) [inline network/src/propagation_network/mod.rs:39]: typo (stop_braodcast)

File: network/src/lib.rs (hunk 1 above)
Comment (junha1) [reply network/src/lib.rs]: fixed

File: network/src/propagation_network/mod.rs (hunk 2 above)
Comment (junha1) [reply network/src/propagation_network/mod.rs]: fixed

**Review by lemonholic — APPROVED** [review]

**Pull request merged**